students/ganyoa/lesson03/mailroom.py

```diff
@@ -0,0 +1,95 @@
+import sys
+
+#main menu; selection will guide the 'main' function
+prompt = "\n".join(("\n""Mailroom options",
+          " Please choose from below options:",
+          " 1 - Send a Thank You",
+          " 2 - Create a Report",
+          " 3 - Quit",
+          ">>> "))
+
+#list of current donor and donation amounts
+donor_list = [("Mouth Devereaux", [4562.58, 3817.39]),
+            ("Mikey Walsh", [618.45, 49.20, 542.87]),
+            ("Andy Carmichael", [75.28, 99.25]),
+            ("Chester Copperpot", [750254.00, 802154.11, 1.75]),
+            ]
+
+
+def average(donor_ave):
+    #calculates donation averages per donor
+    return (sum(donor_ave) / len(donor_ave))
+
+def total_donated(don_amts):
+    #calculates the total donation amount per donor
+    return sum(don_amts[1])
+
+def thank_you():
+    #input option for the user to enter a new name, add amount to previous donor, or see list of previous donors
+    donor_name = input("Enter donors full name (type 'list' to display all names)? > ").title()
+
+    #if user does not enter a name or 'list' they will be brought back to the main menu
+    if donor_name == '':
+        main()
+
+    #displays previous donor list
+    if donor_name == 'List':
+        for i in range(len(donor_list)):
+            print(donor_list[i][0])
+        main()
+    match = 0
+
+    #iterate through donor list, add new name and amount, or amount if the name was already on the list
+    for i in range(len(donor_list)):
+        if donor_name == donor_list[i][0]:
+            match = i
+            break
+    else:
+        donor_list.append((donor_name,[]))
+        match = i + 1
+    donation_amt = input("\n""How much did " + donor_name + " donate? > ")
+    amt = round(float(donation_amt),2)
+    donor_list[match][1].append(amt)
```

Comment:
you can move the amount prompt above your for loop and the donation right in place then you don't need to keep track of index.